klog.go

```diff
@@ -1096,6 +1096,17 @@ func (l *loggingT) setV(pc uintptr) Level {
 	return 0
 }
 
+// Writer implements io.Writer interface as a pass-through for klog.
```

Comment:
is this better than a typedef?

```go
type Writer func(args ...interface{})

func (w Writer) Write(p []byte) (n int, err error) {
  w(string(p))
  return len(p), nil
}
```

or a constructor that masks the concrete type

```go
func NewWriter(logFunc func(args ...interface{})) io.Writer {
   return &writer{logFunc: logFunc}
}
...
```

Reply:
@liggitt I had thought about the constructor, but didn't consider
```type Writer func(args ...interface{})```
which I think might be superior.  I will update with that.

Reply:
the last question I have is whether such an interface belongs in klog. where do we envision using this? given how tiny the adaptation is, would it make sense to gather experience with a few sample uses first before exposing it as an API in klog?

Reply:
@liggitt this is mostly covered in the first comment.  kubectl libraries take io.Writer interfaces for printing messages rather than logging.  In particular, we'd like to re-use the kubectl drain logic elsewhere (in our case, in cluster-api's machine-controller to drain nodes before deleting backing machine objects).

I actually came to klog first expecting this to already exist!  For me, it's a natural place for this code to live rather than potentially duplicating it elsewhere.  It remains to be seen how prolific it's usage might be.

Reply:
I wouldn't have expected to put it here for the following reasons:
* the adapter is not klog specific
* nothing blocks consumers from wrapping it this way
* it is not complex to implement (~2 lines of code plus a type declaration)

if kubectl libraries have a need to use this with IOStreams, it would make more sense to me to place the adapter there

Reply:
I think it should be defined here because I think it might be useful to klog users, wherever those users might be.

It doesn't feel like it belongs in kubectl because kubectl wants to print to stdout/stderr.  It doesn't feel like it belongs in client-go (if some parts of kubectl go there) because client-go won't really have an opinion on where you should log or print things (presumably).  It seems counter intuitive to declare a type/interface for something you don't actually own (the function signature for klog.Info et al is quite generic).

I think klog is a good 'home' for this little piece of code.